### ISSUE TITLE
Improve the Preview Runtime Download Page and Provide Direct Download Links

### ISSUE THREAD

**Issue by RickStrahl** [MicrosoftEdge/WebView2Feedback#722]
The download links here:

https://developer.microsoft.com/en-us/microsoft-edge/webview2/

have no direct links to each of the downloaders. If I have an application and I want to direct people to the download page to a specific installer I can't find a link to do that - users have to choose from the list of installer (which is a crappy page that you have to scroll to get to what you need with no explanations).

The second page (the download and accept page is fine, but you can't link to that because apparently the links are script driven.

Please provide direct links so an application can directly link to one of these downloaders that we want users to install!

If not that then at least make the landing page cleaner and make the links show above the fold.

**Comment by champnic** [comment]
I've added this to our backlog. I agree at the very least the links should be above the fold. We may have some limitations on direct links because of licensing reasons, but I'll see if we can improve that too. Thanks!

**Comment by RickStrahl** [comment]
Hmmm... odd. Trying the 'Get the Link' link and now it's providing me a direct download link which is better. So I could allow downloading automatically into the Downloads folder and then launching the installer. 

I would much prefer this flow:

* Link to the Download Page 
* Link should point to specific installer (ie. Evergreen Bootstrapper)
* Page shows licensing options just as it does now
* Downloads
* User launches install

Not sure if the license page is really necessary - users have to accept the license agreement when the runtime gets installed so I think that should be enough no?

**Comment by RickStrahl** [comment]
@champnic  - Related note. I can't figure out how to install the current runtime (ie. preview runtime) that's **newer** than the runtimes listed on the download page.

How do I even install a newer run that is compatible? 

I had a newer version installed, but uninstalled so I could test what happens when no or an old runtime is installed. But now I can't get the newer version back. 

How do I get the latest preview Runtime? Canary install?

Ultimately is the goal to ship these runtimes as part of the Edge install? Having to install these runtimes seems like a big hassle when the browser is already on the machine. I figured this was what Evergreen mode was supposed to be all about.

**Comment by liminzhu** [comment]
@RickStrahl 

> Can we assume those links are permanent or do they expire since they clearly look like generated Ids?

Yeah we can assume we'll be support the link for a while and if we were to deprecate anything there will be a deprecation process. 

With regard to the website design, we can definitely make improvements to it. I do want to call out that the current download site is only intended for developers, not at all for end users. The license there is specifically the developer license, not end user license, and It'd be very confusing to end users what all these options are. We're working on another site that we or developers can direct end users to, and it should have the end user license instead. 

> Related note. I can't figure out how to install the current runtime (ie. preview runtime) that's newer than the runtimes listed on the download page. How do I even install a newer run that is compatible? I had a newer version installed, but uninstalled so I could test what happens when no or an old runtime is installed. But now I can't get the newer version back. How do I get the latest preview Runtime? Canary install?

Yeah you can use any non-stable Edge channel for development and testing purposes. Changes in general roll up from Canary to Dev to Beta to WebView2 Runtime. 

> Ultimately is the goal to ship these runtimes as part of the Edge install? Having to install these runtimes seems like a big hassle when the browser is already on the machine. I figured this was what Evergreen mode was supposed to be all about.

Not exactly as part of Edge, since it is by design for us to keep the ability the ship Edge/WV2 separately. It's mostly because 1) Enterprise often version-control Edge and we want WebView2 to not affected by the browser, and 2) we want to be able to service any urgent browser/WebView2 fixes without impacting the users of the other world. 

But yeah I do get having to distribute a component is a big pain point. We're working on including WV2 Runtime in-box for future Windows release and best-effort pushing WV2 Runtime to in-market devices. The goal is for WV2 to have more or less similar coverage to Edge on Win10.

**Comment by RickStrahl** [comment]
> With regard to the website design, we can definitely make improvements to it. I do want to call out that the current download site is only intended for developers, not at all for end users. The license there is specifically the developer license, not end user license, and It'd be very confusing to end users what all these options are. We're working on another site that we or developers can direct end users to, and it should have the end user license instead.

I understand, but that's the point of this issue. We need something we can direct people to so:

* We can let people download the runtime  (direct link ideally)
* Have a runtime installer we can bundle as part of an install

The main point I see is this: I want to avoid shipping 90megs of runtime, so I want to use the minimal installer to get it installed. I can do with my installer, but then I presume I have to keep updating this thing with every release which is pretty cumbersome. 

Or I do it as part of my application where I check versions and if out of sync automatically download the installer and re-run the setup which will presumably pull the latest bits.

Either way the runtime installer (the actual app) needs to be end user friendly and should not be a developer tool because at some point we need need to get runtimes installed if not shipping ~100mb with our apps.

My preferred scenario would be to **have the WebView2 runtime ship with Edge installs**. That would be the ideal scenario because that would probably get >95% of users. For the rest the runtime could be manually downloaded using the process above that could be handled during application startup.
